### PR TITLE
Codacy fix new

### PR DESCRIPTION
**Pull request by debasishdebs**
fixing #22

### CODE REVIEW

**Comment by debasishdebs** [comment]
@chupman : I've fixed Codacy errors related to `List Indentation`. Though there are conflicts, I'm unable to resolve them. I used online Github's conflict resolver tool, but once I resolve conflict, and I click option of `commit merge` in UI, it doesn't add any signature, and hence the commit then becomes `cla:no` which we don't want. Any suggestion on ow to handle this?

**Comment by chupman** [comment]
Looks like you need to rebase this branch against initial-pr. Here's an overview of what you need to do with the actual commands to use. 

1. Update all branches with git fetch --all
2. Pull to make sure the initial-pr branch has all the latest commits
3. run a rebase against initial-pr from 
4. resolve any rebase conflicts
5. push the changes

```
git fetch --all
git checkout initial-pr
git pull
git checkout codacy_fix_new
git rebase initial-pr
git status
git add before-script.sh
git add build-docs.sh
git add build-library.sh
git rebase --continue
git push --force <name of your remote fork> codacy_fix_new
```

After you run this you can do a diff against the last commit in initial-pr and the number of files changes should be drastically lower, making my job drastically easier :)
```
git diff 26e217b17946763ef3ac8918f44cee1f5be6b6ac --stat
 BUILDING.md                                       |  2 ++
 README.md                                         |  7 +++----
 build.py                                          |  2 +-
 src/integrationtest/python/DocTraversals_tests.py |  1 -
 src/integrationtest/python/GeoAttribute_tests.py  | 53 +++++++++++++++++++++++++++++------------------------
 5 files changed, 35 insertions(+), 30 deletions(-)
```

File: src/integrationtest/python/DocTraversals_tests.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# Name: Debasish Kanhar
```
Comment (chupman) [inline src/integrationtest/python/DocTraversals_tests.py]: These integration test files should belong to the other pull request. I might be easier to try and merge PR #33 to initial-pr and then do a rebase to clean this up.

File: src/unittest/python/core/datatypes/RelationIdentifier_test.py [hunk 2]
```diff
@@ -30,3 +30,10 @@ def test_relationID_equality(self):
 
         self.assertEqual(e1, e2)
         self.assertNotEqual(e1, e3)
+<<<<<<< HEAD
```
Comment (chupman) [inline src/unittest/python/core/datatypes/RelationIdentifier_test.py]: This definitely needs to get cleaned up.

**Review by chupman — CHANGES_REQUESTED** [review]
Right now this PR appears to be creating more codacy issues than it resolves.  Please shoot me a mention when it's cleaned up and ready for review. If it helps you can enable codacy on your own fork and use it to quickly iterate though fixes without have to deal with the email spams it generates for pull requests.

**Comment by debasishdebs** [comment]
Merged this branch with origin/integration_tests . Resolved a lot of Codacy related issues there, and plan to close/rebase/clean this up based on what should be done, and continue fixing integration_tests (which is done based on reviews) and clean this up

**Comment by FlorianHockmann** [comment]
@debasishdebs I think it makes sense to handle this PR independent of the integration test PR, but if you prefer to base one on the other, that's also OK to me. I would then only review the smaller PR first and the other only after the smaller was merged.

Codacy still complains about a lot of issues in this branch and this PR even adds new issues. Please notify us when you think that this PR is ready for review. If you don't have the time to fix all issues Codacy complains about, then that's also fine of course. Someone else can then still tackle them, but this PR should at least not add new issues. (If you think that Codacy is wrong with some comments, just comment on them and we can see whether we can turn off a Codacy rule.)

**Comment by debasishdebs** [comment]
@FlorianHockmann : I don't know how the comments get missed out. I was thinking to put out a post on dev mailing list asking users to review this. Do you do some sort of subscription or similar so that I don't miss out on comments in future.

As for regarding the PR, I still don't know what can be done here. Codacy always pops up with a new error once I resolve the older error. I don't even know how to test locally before I push changes, Can you guide me in that sense? Because a lot of thigs which Codacy complaints about can be ignored.

You are correct, having so many Codacy issues is a pain for reviewers to review. I have been quite busy with other assignments with JanusGraph especially one related to adding of a new backend namely SnowFlake. 

What would be next course of action you suggest? I also have created another PR #33 which I feel is ready for review. That was primarily targetted for resolving integration test based issues, and integration tests are not passing. Can you review that, and we can decide on course of action on how do we resolve the Codacy issues which we face (Personally I feel 1-2/5 issues are actual valid issues).

**Comment by FlorianHockmann** [comment]
> Do you do some sort of subscription or similar so that I don't miss out on comments in future.

You can either enable notifications for the whole repository by clicking on _Watch_ in the top right corner or subscribe for notifications of just this PR, but that should already be the default in your case as you created the PR. It then depends on your GitHub settings whether you receive notifications per email or not.

> Codacy always pops up with a new error once I resolve the older error.

You probably have to inspect the Codacy comments directly in its dashboard as Codacy only comments on changes in a PR. So if you add another change to fix a Codacy issue, you also give it more code to analyse in which it can find yet more issues.

> I don't even know how to test locally before I push changes, Can you guide me in that sense?

I don't think that you can test Codacy locally, but you can of course also activate it for your own fork and then inspect / fix its issues there before you create a PR if you prefer that.

> Because a lot of thigs which Codacy complaints about can be ignored.

If that's the case, then we can also disable some of Codacys rules to ignore these things. It would be good if you could point to example for the issues that can be ignored in your opinion. So, we can configure Codacy afterwards.

> What would be next course of action you suggest?

I suggest that we now concentrate on getting a basic PR ready and merged to master and therefore ignore this PR for now. You can probably also close this PR. You can still keep the branch in your fork so you don't lose any improvements you made as you can still use them in a follow-up PR in the future again.

**Comment by FlorianHockmann** [comment]
Closing these 3 PRs as we have just merged another PR which contributed a first version for janusgraph-python.
@debasishdebs You are however of course more than welcome to still contribute to this library if you are still interested in that.

**Pull request closed without merging**